src/scripts/term_range_expansion.py

```diff
@@ -0,0 +1,207 @@
+        lambda x: x.replace('http://purl.bioontology.org/ontology/ICD10CM/', 'ICD10CM:'))
+    # # Capture source information
+    # TODO attempt 2
+    xxx = pd.concat([results_df_included_kids, results_df_excluded_kids]).fillna('')
+    xxx['term_info_source'] = 'icd10cm.ttl'
+    # TODO: fix x/y (*Can I re-use `left_join_update()` from valueSetTools?)
+    # df['term_info_source'] = ''
+    # results_df_excluded_kids['term_info_source'] = 'icd10cm.ttl'
+    # results_df_included_kids['term_info_source'] = 'icd10cm.ttl'
+    # # Determine which factors were not in source
+    term_in_icd10cm_ttl = set(xxx['term_id'])
+    df['term_in_source_icd10cm.ttl'] = df['term_id'].apply(
+        lambda term_id: term_id in term_in_icd10cm_ttl)
+
+    # Join
+    # df2 = pd.merge(df, results_df_excluded_kids, how='left', on='term_id')
+    df3 = pd.merge(df, xxx, how='left', on='term_id').fillna('')
+    print(list(df3.columns)) # TODO temp
+
+    # Restructure
+    df4 = df3
+    df4['PK_unique_path'] = \
+        df4.apply(lambda row: row['term_range_id'].replace('ICD10CM:', '') + '_' + row['term_id'].replace(
+            'ICD10CM:', '') + '_' + row['child_id'].replace('ICD10CM:', ''), axis=1)
+    df4['PK_unique_path'] = df4['PK_unique_path'].apply(lambda x: x[:len(x) - 1] if x.endswith('_') else x)
+    df4 = df4.rename(columns={
+        'exclusion_reason': 'child_exclude_reason',
+        'exclude_children': 'child_exclude'})
+    df4 = df4[[
+        'PK_unique_path', 'term_range_id', 'term_range_label', 'term_id', 'term_label', 'child_id',
+        'child_label', 'term_info_source', 'term_in_source_icd10cm.ttl', 'child_exclude', 'child_exclude_reason']]
+    df4 = df4.sort_values(['PK_unique_path', 'term_range_id', 'term_id', 'child_id'])
+
+    # todo: change path
+    df4.to_csv('~/Desktop/results.csv', index=False)
+
+    return df4
+
+    return results
+
+
+def get_parser() -> ArgumentParser:
+    """Add required fields to parser."""
+    package_description = \
+        'Takes in a full ontology and an exclusions TSV and extracts a simple list of terms.\n'\
+        'Outputs two files:\n'\
+        '1. a simple list of terms (config/ONTO_NAME_term_exclusions.txt)\n'\
+        '2. a simple two column file with the term_id and the exclusion reason\n'
+    parser = ArgumentParser(description=package_description)
+
+    parser.add_argument(
+        '-o', '--onto-path',
+        help='Path to the ontology file to query.')
+
+    parser.add_argument(
+        '-e', '--exclusions-path',
+        help='Path to a TSV which should have the following fields: `term_id` (str), `term_label` (str), '
+             '`exclusion_reason` (str), and `exclude_children` (bool).')
+
+    return parser
+
+
+def cli():
+    """Command line interface."""
+    parser = get_parser()
+    kwargs = parser.parse_args()
+    kwargs_dict: Dict = vars(kwargs)
+
+
+    # TODO: temporary hack; need to get .owl versions of these eventually
+    onto_file_remappings = {
```

Comment:
Temporary hack. I have / have been constructing these as RDF `.ttl`. Maybe it's OK to keep using that? But we probably want to standardize everything to the same file extension. So maybe there is some way to get these into `.owl` isntead?

Reply:
will verify that the new make goal works and if so, i can drop this hacky logic